test/verify/check-users

```diff
@@ -546,6 +551,121 @@ class TestAccounts(MachineCase):
         self.allow_journal_messages(".*required to change your password immediately.*")
         self.allow_journal_messages(".*user account or password has expired.*")
 
+    def testCustomUID(self):
+        b = self.browser
+
+        self.login_and_go("/users")
+
+        # Test custom UID
+        createUser(
+            browser=b,
+            user_name="bob",
+            real_name="Bob Bobson",
+            password=good_password,
+            locked=False,
+            force_password_change=False,
+            uid="1500",
+            verify_created=True,
+        )
+        b.wait_visible("#accounts-list td[data-label='Username']:contains('bob') + td + td:contains('1500')")
+
+        # Test dialog predicts corrent next available UID
+        createUser(
+            browser=b,
+            user_name="john",
+            real_name="John Johnson",
+            password=good_password,
+            locked=False,
+            force_password_change=False,
+            expected_uid="1501",
+            verify_created=True,
+        )
+        b.wait_visible("#accounts-list td[data-label='Username']:contains(john) + td + td:contains(1501)")
+
+        # Test creation of users with the same UID
+        createUser(
+            browser=b,
+            user_name="jack",
+            real_name="Jack Jackson",
+            password=good_password,
+            locked=False,
+            force_password_change=False,
+            uid="1501",
+            verify_created=False,
+        )
+        b.wait_visible("#accounts-create-dialog")
+        b.wait_in_text("#accounts-create-user-uid-helper", "already used")
+        b.wait_visible("button.apply:disabled")
+        b.click("button:contains('Create account with non-unique UID')")
+        b.wait_not_present("#accounts-create-dialog")
+        b.wait_in_text("#accounts-list", "Jack Jackson")
+        b.wait_visible("#accounts-list td[data-label='Username']:contains(jack) + td + td:contains(1501)")
+        b.wait_visible("#accounts-list td[data-label='Username']:contains(john) + td + td:contains(1501)")
+
+        # No UID specified
+        createUser(
+            browser=b,
+            user_name="failedcreation",
+            real_name="Failed Creationson",
```

Comment:
I think "Failed Failson" would fit the pattern better :stuck_out_tongue_winking_eye: